File: android/src/main/java/com/aheaditec/freerasp/utils/RandomGenerator.kt
package com.aheaditec.freerasp.utils

import java.security.SecureRandom
import java.util.concurrent.ConcurrentHashMap

internal object RandomGenerator {
    private val secureRandom = SecureRandom()

    private val generatedNumbers = ConcurrentHashMap.newKeySet<Int>()

    internal fun next(): Int {
        val min = 10_000_000
        val max = 999_999_999
        val range = (max - min) + 1

        var nextNumber: Int
        do {
            nextNumber = secureRandom.nextInt(range) + min
        } while (!generatedNumbers.add(nextNumber))

        return nextNumber
    }
}

File: ios/Plugin/utils/RandomGenerator.swift
import Foundation

internal class RandomGenerator { 
    
    private static var generatedNumbers = Set<Int>()
    private static let lock = NSLock()

    internal static func next() -> Int {
        lock.lock()
        defer { lock.unlock() }

        let min = 10_000_000
        let max = 999_999_999

        var nextNumber = Int.random(in: min...max)

        while !generatedNumbers.insert(nextNumber).inserted {
            nextNumber = Int.random(in: min...max)
        }

        return nextNumber
    }
}